include/IPipelineManager.hpp

```diff
@@ -0,0 +1,45 @@
+#ifndef IPIPELINEMANAGER_HPP
+#define IPIPELINEMANAGER_HPP
+
+#include <vector>
+#include <cstddef>
+
+#include "IClient.hpp"
+
+namespace api
+{
+    /**
+     * @interface IPipelineManager
+     * @brief Defines the basic behavior of the pipeline manager.
+     * @tparam RequestDataType Refers to the request data type. It will be transferred from the
+     * parser to the modules.
+     * @tparam ResponseDataType Refers to the request response data type that will be
+     * transferred between the modules and the IReplySender.
+     */
+    template <class RequestDataType, class ResultDataType>
+    class IPipelineManager
+    {
+        public:
+            /**
+             * @brief Starts the pipeline listener when called. If the listener is already started, it should throw an
+             * exception.
+             */
+            virtual void listen() = 0;
+
+            /**
+             * @brief Stops the pipeline listener when called. If the listener is already stopped, it should throw an
+             * exception.
+             */
+            void stop() = 0;
+
+        private:
```

Comment:
Wait, that's useless since inheritance doesn't take private elements from a class it only takes up to protected, so this should be changed to protected